Header header
float32[] generalized_positions